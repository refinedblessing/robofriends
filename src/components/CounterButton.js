import React, { useState } from 'react';

const CounterButton = ({ color }) => {
  const [counter, setCounter] = useState(0);

  return (
    <button
      id="counter"
      color={color}
      onClick={() => setCounter(counter + 1)}
      >
      Count: {counter}
    </button>
  )
}

export default CounterButton;
