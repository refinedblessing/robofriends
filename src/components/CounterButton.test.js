import { shallow } from 'enzyme'
import CounterButton from './CounterButton';
import React from 'react'

it ('enzyme works', () => {
  const mockColor = 'red';
  expect(shallow(<CounterButton color={mockColor}/>).length).toEqual(1)
  expect(shallow(<CounterButton color={mockColor}/>)).toMatchSnapshot();
})

it('correctly implements counter', () => {
  const mockColor = 'red';
  const wrapper = shallow(<CounterButton color={mockColor}/>)

  wrapper.find('#counter').simulate('click')
  expect(wrapper.find('#counter').text()).toEqual('Count: 1')
  wrapper.find('#counter').simulate('click')
  expect(wrapper.find('#counter').text().includes('2')).toBeTruthy()
})