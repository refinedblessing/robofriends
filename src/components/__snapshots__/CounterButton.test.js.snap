// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme works 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
